pyvrp/cpp/DurationSegment.cpp

```diff
@@ -30,11 +30,13 @@ DurationSegment::DurationSegment(ProblemData::Client const &client)
 {
 }
 
-DurationSegment::DurationSegment(ProblemData::VehicleType const &vehicleType)
+DurationSegment::DurationSegment(ProblemData::VehicleType const &vehicleType,
+                                 bool const isStartSegment)
     : duration_(0),
       timeWarp_(0),
-      twEarly_(vehicleType.twEarly),
-      twLate_(vehicleType.twLate),
+      twEarly_(vehicleType.earliestStart),
+      twLate_(isStartSegment ? vehicleType.latestStart
+                             : vehicleType.latestFinish),
```

Comment:
#I'd replace the `isStartSegment` argument with a `Duration twLate` argument. Looking at the places where this constructor is used, we can do (for example)
```cpp
{vehicleType_, vehicleType_.latestStart};
```
instead of
```cpp
{vehicleType_, true};
```
Passing in the relevant data avoids the branch being needed here.